File: promocode-issue/src/main/java/com/voshodnerd/promocode/issue/model/PromoCode.java
package com.voshodnerd.promocode.issue.model;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Data
public class PromoCode {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String value;
    @Enumerated(EnumType.STRING)
    private PromoCodeStatus status;
    @ManyToOne
    @JoinColumn(name = "campaign_id")
    private Campaign campaign;

    public enum PromoCodeStatus {
        ACTIVE,
        USED,
        EXPIRED
    }
}

File: promocode-issue-kotlin/src/main/kotlin/com/voshodnerd/promocode/issue/model/PromoCode.kt
package com.voshodnerd.promocode.issue.model;

import jakarta.persistence.*;

@Entity
class PromoCode {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null
    var value: String = ""

    @Enumerated(EnumType.STRING)
    var status: PromoCodeStatus? = null

    @ManyToOne
    @JoinColumn(name = "campaign_id")
    var campaign: Campaign? = null

    constructor(value: String, status: PromoCodeStatus?, campaign: Campaign?) {
        this.value = value
        this.status = status
        this.campaign = campaign
    }

    enum class PromoCodeStatus {
        ACTIVE,
        USED,
        EXPIRED
    }
}
